src/rdkafka_fetcher.c

```diff
@@ -553,10 +709,16 @@ rd_kafka_fetch_reply_handle_partition(rd_kafka_broker_t *rkb,
 
         if (unlikely(hdr.ErrorCode != RD_KAFKA_RESP_ERR_NO_ERROR)) {
                 /* Handle partition-level errors. */
+                rd_kafka_buf_skip(rkbuf, hdr.MessageSetSize);
+
+                if (rd_kafka_buf_ApiVersion(request) >= 16)
+                        rd_kafka_buf_read_tags(
+                            rkbuf, rd_kafkap_Fetch_reply_tags_partition_parse,
+                            TopicTags, PartitionTags);
+
                 rd_kafka_fetch_reply_handle_partition_error(
                     rkb, rktp, tver, hdr.ErrorCode, hdr.HighwaterMarkOffset);
 
-                rd_kafka_buf_skip(rkbuf, hdr.MessageSetSize);
                 goto done;
```

Comment:
I think we should keep it here and in the done only read if `hdr.ErrorCode` is `NOERROR` because otherwise we'd have to move all the other operations after this which are using rkbuf in done.

Reply:
Reverted it